src/filtration_system/mod.rs

```diff
@@ -38,8 +39,9 @@ impl Decoders {
 pub fn filter_and_get_decoders() -> Decoders {
     trace!("Filtering and getting all decoders");
     let base64 = Base64Decoder::new();
+    let reversedecoder = ReverseDecoder::new();
     Decoders {
-        components: vec![Box::new(base64)],
+        components: vec![Box::new(base64), Box::new(reversedecoder)],
```

Comment:
ToDo -- do I really need to put each one in a box?

Reply:
Yes, because to run rayon at the node level we need this:

```rust
    pub components: Vec<Box<dyn Crack + Sync>>,
```

But since our BFS implementation is tasty we can always just move to using Rayon there and getting rid of this.